### DISCUSSION TITLE
No more devPlugins

### DISCUSSION THREAD

**Discussion by goaaats** [goatcorp/Dalamud#892] [Announcements]
We're planning to remove the "devPlugins" folder with the next major patch. It's primarily used now by devs to distribute testing builds of their plugins, which is contrary to its purpose and causes issues after patches and for troubleshooting, as these builds can never update.

For development, every developer should add their plugin output folder as a developer plugin in the Dalamud settings.
For testing, use the official repo testing mechanism or set up a testing repo.

Thanks!

**Comment by slotthhy** [comment]
Good. The nightmare of support for those leaving stuff in devPlugins is far worse than having developers copy a path into their settings occasionally.

**Comment by KazWolfe** [comment]
While I understand the general purpose for this change, I'm a bit hesitant as I still think there are some legitimate use cases for an easily accessible scratch space. 

One such common use case I've seen (and have used in the past) is distributing extremely rapid-fire changes between developers trying to test something on a machine that isn't theirs. For example, after running into a hard-to-reproduce bug, a developer just sent me a debug build and asked for output. This resulted in another build, and then yet another one as the bug was isolated. This sort of rapid-fire testing strategy doesn't exactly fit with the model of testing repos, nor do I suspect it would work too effectively replacing files in `installedPlugins`. It definitely is possible to add a temp workspace to the dev plugin locations list, but it does become a minor annoyance at times (especially for tests that take all of ten minutes).

**Reply by philpax** [reply]
DIP17 should give us nice zip artifacts that should be much easier for people to load in (no creating a folder, you just pick the zip in the settings - potentially with a nice file picker.) Does that alleviate your concerns?

**Reply by KazWolfe** [reply]
As long as there's a fast way I can get/send artifacts to others as an extension of my own dev process, I think that's fine. Basically, I used the scratch space for the following two cases with other people I knew and trusted as devs:

- "Hey, I have no idea if this fixes your bug, can you try and let me know?"
- "I need more info - can you load this version and send me the log file?"

A zip would work for these cases, and I can just instruct the user to remove the file after they're done.

**Reply by MidoriKami** [reply]
> DIP17 should give us nice zip artifacts that should be much easier for people to load in (no creating a folder, you just pick the zip in the settings - potentially with a nice file picker.) Does that alleviate your concerns?

That would actually be super nice, especially if it unloaded the old zip when given a new one with the same plugin name.

Also, ideally it would auto-unload the plugin on restart.

**Comment by philpax** [comment]
As mentioned in Discord: I think we should not do this until #874, so that we can offer people aggrieved a meaningful alternative that makes up for some of the reasons why people continue to use devPlugins today.

**Comment by avafloww** [comment]
This change has been completed with the merge of #1219.
